src/client/sandbox/event/selection.js

```diff
@@ -104,7 +122,15 @@ export default class Selection {
     }
 
     static _needChangeInputType (el) {
-        return domUtils.isInputElement(el) && browserUtils.isWebKit && /^(number|email)$/.test(el.type);
+        return (browserUtils.isWebKit || Selection._needForInternalSelection())
+               && domUtils.isInputElement(el) && /^(number|email)$/.test(el.type);
+    }
+
+    // NOTE: input elements in Firefox since version 51 are needing
+    // for alternative selection properties, because original are
+    // clearing while input type changes to url, email etc
```

Comment:
Something like: `we need to store element state because it's cleared when it's type is changed`.